pkg/stats/monitor.go

```diff
@@ -168,33 +188,78 @@ func (m *Monitor) GetRequestCount() int {
 	return int(m.requests.Load())
 }
 
+func (m *Monitor) UpdatePID(egressID string, pid int) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	ps := m.pending[egressID]
+	delete(m.pending, egressID)
+
+	if existing := m.procStats[pid]; existing != nil {
+		ps.maxCPU = existing.maxCPU
+		ps.totalCPU = existing.totalCPU
+		ps.cpuCounter = existing.cpuCounter
+	}
+	m.procStats[pid] = ps
+}
+
+func (m *Monitor) GetUsageStats(egressID string) (float64, float64) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	for pid, ps := range m.procStats {
+		if ps.egressID == egressID {
+			delete(m.procStats, pid)
```

Comment:
why do we delete it here? seems like a `GetXX` should not have a side effect like this?

Reply:
I'll rename, it's only called after the egress ends